### ISSUE TITLE
Link directly to source file on GitHub

### ISSUE THREAD

**Issue by thomasheartman** [enonic/site-developer#412]
With the new feedback links (#328, #411) in place, a potential improvement would be to link directly to the source file that you're viewing, so that you can edit a document directly on GitHub if you find something.

This may require some work, however, as we need to link to the direct source file. To my knowledge, this data is not available in the content model at the moment.

**Comment by thomasheartman** [comment]
@alansemenov @sigdestad We discussed this briefly in the sprint review/planning today, and I agree that being able to link directly to the source document would be a great feature! I also had a quick idea session with @poi33 about this after the planning.

To get started on this, however, I need some more information, and I think you might be able to help me out: how can I find out what file a page is generated from?

The main issue is that there doesn't seem to be any data relating to what the source file of a document/page is. That is, there is no easy way to find out what document generated https://developer.enonic.com/guides/introduction-to-enonic/xp7/deployment, for instance. 

In many cases, the url matches the document name, but not necessarily the path. With the above example, the path (in the repo) would be `docs/deployment.adoc` and the relevant GitHub link would be https://github.com/enonic/doc-intro-to-enonic/blob/master/docs/deployment.adoc.

If you assume that all repos are structured the same, this can be worked around fairly easily. However, for guides such the one for lib-static and the Gatsby starter, the path is `docs/index.adoc`. @poi33 also suggested that there may be even more variations to watch out for.

While it might be possible to write code to guess and to handle this with the current data, it would be much easier if we could make this directly accessible.  Can we somehow expose more data about the repo in some of the models, including what source file it's built from?

**Comment by sigdestad** [comment]
With the exception of the "root" page, we don't use index.html - so you should be able to determine the name of the page as follows:
/ -> index.adoc
/mypage -> mypage.adoc
/mypage/whatever -> mypage/whatever.adoc

prefix this with repo-url/blob/master/docs/ and you should be fine: NB! For multi-version docs, there are actually different branches being used. We could solve this by only linking to edit on github for the "current" version imho.

**Comment by thomasheartman** [comment]
Right. I'll look into it and see what I can find. Still, it sounds like this could easily break, and I wonder if it would be better to include this data directly if that's possible? 

And yeah, for multi-version docs we could either link all to the repo's main branch, or add some version info. How do the multi-version docs work? I noticed [doc-explorer](https://github.com/enonic/doc-explorer) has two versions, but I don't understand how they're deployed individually.

Side note about doc-explorer: I noticed that if you access the doc without a version (https://developer.enonic.com/docs/explorer), you get sent to the 1.4.0 version instead of the stable version. Is that intentional?

**Comment by sigdestad** [comment]
Versions.json controls this. In the case of explorer, 1.4 is then current, and someone probably gave the other one a bad name :)

**Comment by thomasheartman** [comment]
Okay. From what I can tell, 1.4.0 is using a commit from last July, while the stable doc is just based off master (and was last updated five days ago), so yeah, I suppose 'latest' might be a better name for it 🤷🏾 

What do you think about making more data about the repo accessible within site-developer, such as which files were used to generate pieces of content? Is it possible? Impossible? Impractical? Good idea? Bad idea?

**Comment by sigdestad** [comment]
Lets try the simple approach outlined by me first.

**Comment by sigdestad** [comment]
Then the versions.json might be wrong?

**Comment by thomasheartman** [comment]
> Lets try the simple approach outlined by me first.

Sure thing! 

> Then the versions.json might be wrong?

Err, it might be, but it might also not be. I don't have the context to say one way or the other 🤷🏾  From looking at the git history, though, the commit referenced in the versions file coincides roughly with release 1.0.0. However, the reference to that commit (and the creation of the 1.4.0 version) was only created about a week ago. So I reckon it's under control.
